### PR TITLE
[newton] RQE-237 Set ironic gating variables

### PR DESCRIPTION
**Pull request by mkam**
Set scenario variables for ironic and
nova_virt_type for when the scenario
does not include ironic.

Issue: [RQE-237](https://rpc-openstack.atlassian.net/browse/RQE-237)

### CODE REVIEW

**Comment by mkam** [comment]
recheck_cit_trusty_deploy_swift

**Comment by mkam** [comment]
recheck_cit_trusty_deploy_swift

**Comment by mkam** [comment]
Successful run with `ironic-multihypervisor` as scenario: https://rpc.jenkins.cit.rackspace.net/view/Ironic/job/RPC-AIO_newton-xenial-deploy-ironic-multihypervisor-periodic/15/

File: scripts/bootstrap-aio.yml [hunk 1]
```diff
@@ -152,6 +151,32 @@
...
+    - name: Set specific user variables for all scenarios but ironic
+      set_fact:
+        scenario_user_variables_overrides_non_ironic:
+          nova_virt_type: qemu
+
+    - name: Add non-ironic scenario user vars to user_variables_overrides
+      set_fact:
+        user_variables_overrides: "{{ user_variables_overrides|combine(scenario_user_variables_overrides_non_ironic)}}"
+      when: (SCENARIO is defined and SCENARIO != "ironic-multihypervisor") or
+            (SCENARIO is undefined)
```
Comment (mattt416) [inline scripts/bootstrap-aio.yml]: Is this getting set correctly?  Currently we set `nova_virt_type=qemu` when we have a `JOB_NAME` env variable, meaning the job was instantiated through Jenkins.  This doesn't seem to match the current when.

**Comment by mkam** [comment]
recheck_cit_xenial_deploy_ironic-multihypervisor

**Pull request merged**